runtime/interpreter/value.go

```diff
@@ -5562,31 +5562,33 @@ func (v *CompositeValue) InitializeFunctions(interpreter *Interpreter) {
 	v.Functions = interpreter.typeCodes.CompositeCodes[v.TypeID()].CompositeFunctions
 }
 
-func (v *CompositeValue) OwnerValue() OptionalValue {
+func (v *CompositeValue) OwnerValue(interpreter *Interpreter) OptionalValue {
 	if v.Owner == nil {
 		return NilValue{}
 	}
 
+	// There's no direct access to the host environment at this point.
+	// Therefore, load and invoke the `getAccount` host function.
+	getAccountValue, ok := interpreter.effectivePredeclaredValues["getAccount"]
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	getAccountHostFunc := getAccountValue.ValueDeclarationValue().(HostFunctionValue)
```

Comment:
This works, but is brittle, as we will break this accidentally if we e.g. change the name or signature in the future.

Also, currently the `getAccount` function is only injected by the runtime, i.e. outside of the interpreter, so we shouldn't introduce an (implicit) cyclic dependency here.

Given that the `owner` field is baked into the language, and both the `sema` and `interpreter` packages are aware of the notion of a resource having an owner account, I think the proper way to fix this is to make `getAccount` a built-in function too, and make the implementation of it an interpreter option, i.e a handler function that returns a public account value for an address (just like the runtime already does in `interpreterRuntime.newGetAccountFunction`).

I can add this if you want to

Reply:
oh wow, I was thinking all day on a way to eliminate that dependency loop. Only managed to push it to runtime. Didn't know this is even possible! Thank you so much!

Added the interpreter option in https://github.com/onflow/cadence/pull/770/commits/6978995a52f76ea14e98aa79b642f38053c60438

Reply:
Nice, looks great!